include/voxelized_geometry_tools/ros_interface.hpp

```diff
@@ -401,7 +457,9 @@ inline SignedDistanceFieldMessage GetMessageRepresentation(
     const SignedDistanceField<BackingStore>& sdf)
 {
   SignedDistanceFieldMessage sdf_message;
+#if VOXELIZED_GEOMETRY_TOOLS__SUPPORTED_ROS_VERSION == 1
   sdf_message.header.stamp = ros::Time::now();
+#endif
```

Comment:
There's no such thing as a global clock in ROS 2, and I wanted the API to stay the same. Thus, in a ROS 2 build, these messages will not be stamped.

I'd argue that a function that builds a message representation of an abstraction that doesn't involve the concept of time shouldn't be concerned with timestamps, but reviewers may disagree.

Reply:
Yeah, ideally, the stamp should be up to the end-producer of these messages; is there any easy / stupid way to iterate over the nested marker setup to stamp them? e.g.

```
void SetStampRecursive(Container& msg, Time value) {
  msg.header.stamp = value;
  for (auto& item : msg.items) SetStampRecursive(item, value);
  // ...
}
```

Reply:
- Setting message timestamps was a decision I made a long time ago in a predecessor codebase, but I don't think it was the right one. I would go ahead and remove all timestamp setting.

- Either in here, `common_robotics_utilities`, or somewhere else we could easily have helpers to set stamps. The two most common cases to handle:
```
template <typename MessageType, typename Container = std::vector<MessageType>>
void SetMessageTimestamps(Container& messages, const ros::Time& timestamp)
{
  for (auto& message : messages)
  {
    message.header.stamp = timestamp;
  }
}

void SetMessageTimestamps(visualization_msgs::MarkerArray& markers, const ros::Time& timestamp)
{
  SetMessageTimestamps<visualization_msgs::Marker>(markers.markers, timestamp);
}
```

Reply:
Do you agree we defer that additional API to a follow-up PR? Or would you rather add that API first? If so, where? I don't have a strong preference.

Reply:
See https://github.com/calderpg/common_robotics_utilities/pull/9 and https://github.com/calderpg/voxelized_geometry_tools/pull/7. I'll rebase this PR after those go in.

Reply:
For now let's add it here in this PR, and then we can decide later if it should go somewhere else.

I would say the ideal solution would be a codegen'd `SetStampsRecursive` method on every message type to set the stamp on the message and any contained message types, although that seems like a long shot.